arch/xtensa/core/fatal.c

```diff
@@ -92,8 +98,12 @@ void z_xtensa_fatal_error(unsigned int reason, const z_arch_esf_t *esf)
 	if (esf) {
 		z_xtensa_dump_stack(esf);
 	}
-
-	z_fatal_error(reason, esf);
```

Comment:
`z_fatal_error()` does not always halt the system. Applications can implement their own `k_sys_fatal_error_handler()` to override the default behavior. So you need to keep this and remove the `arch_system_halt()` in new code.

Reply:
I dropped the backtrace commit from this PR and will fix in another PR.